docs/design/sql/01-type-system.md

```diff
@@ -127,22 +134,23 @@ converted to the target type.
 
 *Table 4: Type conversions (I - implicit, E - explicit)*
 
-| From/To | VARCHAR | BOOLEAN | TINYINT | SMALLINT | INT | BIGINT | DECIMAL | REAL | DOUBLE | DATE | TIME | TIMESTAMP | TIMESTAMP W/ TZ | OBJECT |
-|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|
-| **VARCHAR** | `I` | `I` | `I` | `I` | `I` | `I` | `I` | `I` | `I` | `I` | `I` | `I` | `I` | `I` |
-| **BOOLEAN** | `E` | `I` |  |  |  |  |  |  |  |  |  |  |  | `I` |
-| **TINYINT** | `E` |  | `I` | `I` | `I` | `I` | `I` | `I` |`I`  |  |  |  |  | `I` |
-| **SMALLINT** | `E` |  | `E` | `I` | `I` | `I` | `I` | `I` | `I` |  |  |  |  | `I` |
-| **INT** | `E` |  | `E` | `E` | `I` | `I` | `I` | `I` | `I` |  |  |  |  | `I` |
-| **BIGINT** | `E` |  | `E` | `E` | `E` | `I` | `I` | `I` | `I` |  |  |  |  | `I` |
-| **DECIMAL** | `E` |  | `E` | `E` | `E` | `E` | `I` | `I` | `I` |  |  |  |  | `I` |
-| **REAL** | `E` |  | `E` | `E` | `E` | `E` | `E` | `I` | `I` |  |  |  |  | `I` |
-| **DOUBLE** | `E` |  | `E` | `E` | `E` | `E` | `E` | `E` | `I` |  |  |  |  | `I` |
-| **TIME** | `E` |  |  |  |  |  |  |  |  |  | `I` | `I` | `I` | `I` |
-| **DATE** | `E` |  |  |  |  |  |  |  |  | `I` |  | `I` | `I` | `I` |
-| **TIMESTAMP** | `E` |  |  |  |  |  |  |  |  | `E` | `E` | `I` | `I` | `I` |
-| **TIMESTAMP W/ TZ** | `E` |  |  |  |  |  |  |  |  | `E` | `E` | `E`  | `I` | `I` |
-| **OBJECT** | `E` | `E` | `E` | `E` | `E` | `E` | `E` | `E` | `E` | `E` | `E` | `E` | `E` | `I` |
+| From/To | NULL | VARCHAR | BOOLEAN | TINYINT | SMALLINT | INTEGER | BIGINT | DECIMAL | REAL | DOUBLE | DATE | TIME | TIMESTAMP | TIMESTAMP W/ TZ | OBJECT |
+|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|
+| **NULL** | `I` | `I` | `I` | `I` | `I` | `I` | `I` | `I` | `I` | `I` | `I` | `I` | `I` | `I` | `I` |
+| **VARCHAR** |  | `I` | `I` | `I` | `I` | `I` | `I` | `I` | `I` | `I` | `I` | `I` | `I` | `I` | `I` |
+| **BOOLEAN** |  | `E` | `I` |  |  |  |  |  |  |  |  |  |  |  | `I` |
+| **TINYINT** |  | `E` |  | `I` | `I` | `I` | `I` | `I` | `I` |`I`  |  |  |  |  | `I` |
+| **SMALLINT** |  | `E` |  | `E` | `I` | `I` | `I` | `I` | `I` | `I` |  |  |  |  | `I` |
+| **INTEGER** |  | `E` |  | `E` | `E` | `I` | `I` | `I` | `I` | `I` |  |  |  |  | `I` |
+| **BIGINT** |  | `E` |  | `E` | `E` | `E` | `I` | `I` | `I` | `I` |  |  |  |  | `I` |
+| **DECIMAL** |  | `E` |  | `E` | `E` | `E` | `E` | `I` | `I` | `I` |  |  |  |  | `I` |
+| **REAL** |  | `E` |  | `E` | `E` | `E` | `E` | `E` | `I` | `I` |  |  |  |  | `I` |
+| **DOUBLE** |  | `E` |  | `E` | `E` | `E` | `E` | `E` | `E` | `I` |  |  |  |  | `I` |
+| **TIME** |  | `E` |  |  |  |  |  |  |  |  |  | `I` | `I` | `I` | `I` |
+| **DATE** |  | `E` |  |  |  |  |  |  |  |  | `I` |  | `I` | `I` | `I` |
+| **TIMESTAMP** |  | `E` |  |  |  |  |  |  |  |  | `E` | `E` | `I` | `I` | `I` |
+| **TIMESTAMP W/ TZ** |  | `E` |  |  |  |  |  |  |  |  | `E` | `E` | `E`  | `I` | `I` |
+| **OBJECT** |  | `E` | `E` | `E` | `E` | `E` | `E` | `E` | `E` | `E` | `E` | `E` | `E` | `E` | `I` |
```

Comment:
- The order of TIME and DATE is different in columns and rows.
- We can use `-` instead of `I` on the diagonal where no conversion takes place (like VARCHAR to VARCHAR)
- On line 132 above: also implicit conversion can fail

Reply:
fixed in fixed in 3c4f1a091791e4047d14c23e9372c3da25561b37